src/main/java/core/basesyntax/operation/ReturnOperation.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+
+public class ReturnOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        FruitStorage.fruitRepository.merge(fruitTransaction.getFruit(),

```

Comment:
It's good practice to ensure that operations like merge on shared resources are done in a thread-safe manner to avoid potential concurrency issues. Consider using ConcurrentHashMap or handling synchronization if needed.